tests/testing.h

```diff
@@ -36,7 +36,7 @@
 #include <sstream>
 #include <functional>
 
-#ifdef _WIN32
+#if (defined _WIN32) || (defined SDL2PP_XCODE_GENERATED)
```

Comment:
I investigated if there is a macro specific for XCode but I couldn't find any. There are some OS X specific macros (like _\_\_APPLE\_\__) but that would completely disable colors in OS X which work if you run the tests on terminal:
![screenshot](https://user-images.githubusercontent.com/493889/28153395-023bf752-675a-11e7-8786-b0973d7ad209.png)

Also I want to note that this will disable the colors if you compile in XCode and that compiled version in terminal (upper screenshot was done without this commit). If you just run `cmake .` without generating a XCode project the tests will be colored.

Other option would be to add an argument _testing.h_'s main to switch the colors and I could pass the argument via XCode but that sounds a lot more of a hassle, I wanted to do a small commit to fix the little things that happen when using XCode.